x-pack/legacy/plugins/apm/public/components/shared/TransactionBreakdown/TransactionBreakdownGraph/index.tsx

```diff
@@ -28,33 +25,39 @@ const TransactionBreakdownGraph: React.FC<Props> = props => {
   >['series'] = useMemo(() => {
     return timeseries.map(timeseriesConfig => {
       return {
-        title: timeseriesConfig.name,
-        color: timeseriesConfig.color,
-        data: timeseriesConfig.values,
-        type: 'area',
-        hideLegend: true
+        ...timeseriesConfig,
```

Comment:
This should be the only transformation left client-side. JSON does not support `undefined` as a value so we cannot transform that server-side.

Reply:
Is this transformation still necessary after you added the `isNaN(d.y)` check to `getNull`?
Maybe there are other things at play here but this codesandbox at least seems to work fine with both `{ x: 1 }` and `{ x: 1, y: undefined}`

https://codesandbox.io/s/6kv44

Reply:
Yeah, the chart got really weird when I tried that. Presumably because we use a lot of `'y' in coord` checks. I'll try it again though, maybe it was something else that is now fixed.